File: epi_judge_java_solutions/epi/OrdinaryMazeGame.java

package epi;

public class OrdinaryMazeGame extends MazeGame {
  // The default constructor will call superclass constructor.
  @Override
  protected Room makeRoom() {
    return new OrdinaryRoom();
  }
}

class OrdinaryRoom implements Room {
  public void connect(Room that) {}
}

File: epi_judge_python_solutions/ordinary_maze_game.py
from maze_game import MazeGame, Room


class OrdinaryMazeGame(MazeGame):
    def make_room(self):
        return OrdinaryRoom()


class OrdinaryRoom(Room):
    def connect(self, room2):
        print('Connecting ordinary room')


x = OrdinaryMazeGame()
assert x is not None
assert x.make_room() is not None
